src/plugins/hetero/src/subgraph_collector.cpp

```diff
@@ -212,7 +212,7 @@ void ov::hetero::SubgraphCollector::split_subgraphs_by_parameter_results() {
     for (const auto& input : ordered_subgraph_inputs) {
         if (!is_graph_input_node(input.get_node())) {
             auto input_source_output = input.get_source_output();
-            if (!is_graph_input_node(input_source_output.get_node())) {
+            if (!ov::op::util::is_constant(input_source_output.get_node())) {
```

Comment:
will this also fix the 2 same params with different ptr issue?

Reply:
also, I think maybe we don't need this for constant as well?

Reply:
@vurusovs  can you please take a look at this?
it solves issue with graph like this:
<img width="464" alt="image" src="https://github.com/openvinotoolkit/openvino/assets/32348500/999e6a6d-8c87-4d9e-8928-282b86f06439">

Reply:
> will this also fix the 2 same params with different ptr issue?

Yes.